### PR TITLE
feat(dapp-console): improve error handling on faucet claim flow

### PR DESCRIPTION
**Pull request by tremarkley**
Closes https://github.com/ethereum-optimism/ecosystem/issues/456

Improves the faucet claim dialog to properly display the claim failed state if the claim tx fails. This will prevent the claim spinner from endlessly spinning when claim txs fail.

### CODE REVIEW

**Comment by tremarkley** [comment]
* **#457** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/457?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tremarkley and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dapp-console/app/hooks/useFaucetVerifications.ts [hunk 1]
```diff
@@ -17,7 +17,10 @@ const useFaucetVerifications = () => {
       {
         address: walletAddress,
       },
-      { enabled: !!walletAddress && !!authenticated },
+      {
+        enabled: !!walletAddress && !!authenticated,
+        refetchOnWindowFocus: false,
```
Comment (tremarkley) [inline apps/dapp-console/app/hooks/useFaucetVerifications.ts:22]: unrelated fix in order to prevent putting the faucet into a loading state when it is not necessary

File: apps/dapp-console/app/faucet/components/ClaimButton.tsx [hunk 2]
```diff
@@ -97,16 +97,21 @@ const ClaimButton = forwardRef<HTMLButtonElement, ClaimButtonProps>(
...
         let response
         if (isOnchainClaim) {
           response = await handleOnchainClaim()
         } else {
           response = await handleOffchainClaim()
         }
 
+        if (response && response.error) {
+          console.error('Claim failed', response.error)
+          onFailed()
```
Comment (nitaliano) [inline apps/dapp-console/app/faucet/components/ClaimButton.tsx:110]: nit: it could be good to return under this line incase we ever have extra logic below this conditional that isn't expecting it to have failed

**Review by nitaliano — APPROVED** [review]

File: apps/dapp-console/app/faucet/components/ClaimButton.tsx (hunk 2 above)
Comment (tremarkley) [reply apps/dapp-console/app/faucet/components/ClaimButton.tsx:110]: oh yeah, good catch
